session19/src/ex5/Main.java

```diff
@@ -0,0 +1,21 @@
+package ex5;
+
+public class Main {
+
+    public static void main(String[] args) {
+        Product product1 = new Product("Laptop", 999.99, "12345");
+        Product product2 = new Product("Phone", 599.99, "12345");
+        Product product3 = new Product("Tablet", 299.99, "67890");
+
+        System.out.println("Product1 equals product2: " + product1.equals(product2));
+        System.out.println("Product1 equals product3: " + product1.equals(product3));
+
+        System.out.println("Hashcode of product1: " + product1.hashCode());
+        System.out.println("Hashcode of product2: " + product2.hashCode());
+
+        product2 = new Product("Phone", 599.99, "54321");
```

Comment:
instead of re-assigning a new Product object you could create a setter for `id` in `Product` class and set the new id here. Something like:
`product2.setId("54321");`